src/Request.hs

```diff
@@ -73,11 +73,13 @@ mkValidator contractInfo@ContractInfo{..} dat lenderTn ctx = validate
       UpperBound (Finite x) _ -> Just x
       _                       -> Nothing
 
-    valueToBorrower :: Value
-    valueToBorrower = valuePaidTo (U.info ctx) (unPaymentPubKeyHash $ borrowersPkh dat)
+    valueToBorrower :: Maybe Value
+    valueToBorrower = fmap (valuePaidTo (U.info ctx)) (toPubKeyHash $ borrowersAddress dat)
```

Comment:
This does not verify the staking part. This takes the `borrowersAddress`, extracts just the payment part from it and checks basically the same logic as before.

The definition of `toPubKeyHash` is this:
```
toPubKeyHash :: Address -> Maybe PubKeyHash
toPubKeyHash (Address (PubKeyCredential k) _) = Just k
toPubKeyHash _                                = Nothing
```
As you can see, it ignores the staking part completely.

I suggest to not use `valuePaidTo` function and implement a similar stronger function `valuePaidToAddress` instead. The function would compare the whole address and not just the payment part of the address. You can put the function in Utils to potentially reuse in future development

Reply:
https://github.com/aadafinance/aada-finance/pull/32/commits/07c4f47a6fd4f2b5b565943141e0648b53bbfbad